core/server/proxy/src/main/java/alluxio/proxy/s3/S3ObjectTask.java

```diff
@@ -912,7 +902,7 @@ public Response continueTask() {
                     .putAllXattr(xattrMap)
                     .setXattrPropStrat(XAttrPropagationStrategy.LEAF_NODE)
                     .build()
-            );
+            ).close();
```

Comment:
which fileinstream? u mean FileOutStream generated by createFile
and do 
try(FileOutStream fos = createfile()) {} 
with an empty block?

Reply:
actually, I think you are good. if you are creating an empty file here, right?

Reply:
yeah so this will be closed by digestOutputStream. and I checked if createFile when opening FileOutStream if any exception it will force close any resources that might be opened, so as long as nothing in between in getting the outstream + use the stream, but i think try-with-resources should be best in that sense. also I was wondering why creating empty file needs to acquire BlockWorkerClientResource instead of acquiring on first write like in getNextBlock? We r actually leaving this close responsibility to the caller right? that's prone to leak